sharktank/sharktank/types/tensors.py

```diff
@@ -801,10 +822,19 @@ def __init__(
         ts: list[torch.Tensor],
         name: str = UnnamedTensorName,
         shape: Optional[list[int]],
+        devices: Tuple[int] | None,
+        pinned: bool | None,
     ):
         assert len(ts) > 0
         assert shard_dim is None or (shard_dim >= 0 and len(ts[0].shape) > shard_dim)
-        super().__init__(name=name, shape=shape, shard_dim=shard_dim)
+        if devices is None:
+            devices = tuple(range(len(ts)))
+        assert len(ts) == len(devices)
+        if pinned is None:
+            pinned = not isinstance(ts[0], torch._subclasses.fake_tensor.FakeTensor)
+        super().__init__(
+            name=name, shape=shape, shard_dim=shard_dim, devices=devices, pinned=pinned
+        )
         self._shards: tuple[DefaultPrimitiveTensor] = tuple(
```

Comment:
How come self._shards is being set here rather than inside `ShardedTensor.__init__`? Moving looks like it would simplify things both for where this PR needs to set `.pinned` and in terms of the repetition for the other subclasses.

Reply:
Probably should have been and was overseen before review. Lets avoid fixing it in this PR as its already quite large.

Reply:
I'll open up a separate PR to clean that up.